examples/births-deaths/population_manager.rs

```diff
@@ -0,0 +1,345 @@
+use crate::parameters_loader::Parameters;
+use ixa::{
+    context::Context,
+    define_derived_property, define_person_property, define_person_property_with_default,
+    global_properties::ContextGlobalPropertiesExt,
+    people::{ContextPeopleExt, PersonId, PersonProperty},
+    random::{define_rng, ContextRandomExt},
+};
+use serde::Deserialize;
+
+define_rng!(PeopleRng);
+
+static MAX_AGE: u8 = 100;
+use rand_distr::Exp;
+use serde::Serialize;
+use std::fmt;
+
+#[derive(Debug, Hash, Eq, PartialEq, Clone, Copy, Serialize, Deserialize)]
+pub enum InfectionStatus {
+    S,
+    I,
+    R,
+}
+
+#[derive(Deserialize, Serialize, Copy, Clone, PartialEq, Eq, Debug, Hash)]
+pub enum AgeGroupRisk {
+    NewBorn,
+    General,
+    OldAdult,
+}
+
+impl fmt::Display for AgeGroupRisk {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{self:?}")
+    }
+}
+
+define_person_property_with_default!(InfectionStatusType, InfectionStatus, InfectionStatus::S);
+
+define_person_property!(Age, u8);
+define_person_property_with_default!(Alive, bool, true);
+define_derived_property!(AgeGroupFoi, AgeGroupRisk, [Age], |age| {
+    if age <= 1 {
+        AgeGroupRisk::NewBorn
+    } else if age <= 65 {
+        AgeGroupRisk::General
+    } else {
+        AgeGroupRisk::OldAdult
+    }
+});
+
+fn schedule_aging(context: &mut Context, person_id: PersonId) {
+    if context.get_person_property(person_id, Alive) {
+        let prev_age = context.get_person_property(person_id, Age);
+        context.set_person_property(person_id, Age, prev_age + 1);
+        let next_age_event = context.get_current_time() + 365.0;
+        context.add_plan(next_age_event, move |context| {
+            schedule_aging(context, person_id);
```

Comment:
Isn't it kind of weird that everyone in the simulation gets a birthday of t = 0 (well, -365*Age) except people who are born after? I defer to you on what is more natural on the simulation side of things but I'd probably do this a bit differently....

Rather than scheduling aging events for each person, I'd probably index by birthday and then check who's age needs to be incremented. Alternatively, you could assume everyone's birthday is t=0 and just increment everyone's age every 365 days. What do you think?

Reply:
I modified this by adding a random birthday from 0 to 365 when creating a new person at the beginning of the simulation. Let me know if this looks good for you. I addressed @ChiragKumar9 comments as well but let me know if I missed anything.